app/code-area/page.tsx

```diff
@@ -25,69 +26,86 @@ const CodeArea = () => {
     );
     const router = useRouter();
     const { mosaicThemeDark } = useCodeAreaStore();
+    const { email } = useStepperStore();
+    const { check } = useStepperStore();
+    const { progress } = useStepperStore();
+    const { challenge } = useStepperStore();
 
-    return (
-        <div className='h-screen'>
-            <div className='pointer-events-auto absolute inset-0 z-20 bg-white bg-opacity-50 xl:hidden' />
-            <div className='absolute inset-0 z-30 flex items-center justify-center xl:hidden'>
-                <div className='w-3/4 max-w-lg rounded-xl bg-black p-6 shadow-lg'>
-                    <h2 className='mb-4 text-center text-xl font-bold text-white'>
-                        It looks like you&quot;re on a small screen
-                    </h2>
-                    <p className='text-sm font-semibold text-gray-400'>
-                        We hate to be non-inclusive towards phones and tablets,
-                        however we want to provide you with the best experience
-                        possible!
-                    </p>
-                    <div className='flex justify-center'>
-                        <Button
-                            variant='destructive'
-                            className='mt-2'
-                            onClick={() => router.push("/")}
-                        >
-                            Back to main page...
-                        </Button>
+    if (
+        email === "" &&
+        check === false &&
+        progress !== (99 || 100) &&
+        challenge === ""
```

Comment:
These should be OR clauses, since either one of these being true should make coming to /code-page illegal. 

Further, this validation is likely needed to be reused elsewhere too so it's a good idea to refactor this.

Reply:
changed